### PR TITLE
phase3: update weave CNI plugin to 2.4.0

### PR DESCRIPTION
**Pull request by neolit123**
1.9.4 is quite outdated and is causing errors for k8s 1.12.
Use weave 2.4.0 and also consolidate the JSON files into a single
List object.

The 2.4.0 manifest is based on:
  curl -s "https://cloud.weave.works/k8s/net.json?k8s-version=v1.10.0" | \
  jq . > phase3/weave-net/weave-net-list.json

Also bump the kubectl version to 1.11.2 as 1.6.x is ancient.

ref: https://github.com/kubernetes/kubernetes/issues/68475

notes:
- this is against the `kubeadm-e2e` branch.
- i have no means to tests this because phase1 fails for me on GCE with unrelated terraform errors that i can't figure out.

/kind bug
/assign @errordeveloper
/cc @timothysc @bboreham

### CODE REVIEW

File: phase3/weave-net/weave-net-list.json [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+            "subjects": [
+                {
+                    "kind": "ServiceAccount",
+                    "name": "weave-net",
+                    "namespace": "kube-system"
+                }
+            ]
+        },
+        {
+            "apiVersion": "rbac.authorization.k8s.io/v1beta1",
```
Comment (timothysc) [inline phase3/weave-net/weave-net-list.json]: v1beta1 ?

**Comment by timothysc** [comment]
@roberthbailey - PTAL this is critical atm. 
/lgtm
/approve

File: phase3/weave-net/weave-net-list.json (hunk 1 above)
Comment (neolit123) [reply phase3/weave-net/weave-net-list.json]: i'm not sure. just downloaded and converted to JSON from YAML.

**Comment by neolit123** [comment]
i also need someone to double check the change in terms of jsonnet and `kubeadm -f apply <folder>`.
it _should work_, by just having a file with `List` object in that folder instead of the separate object files.

**Comment by neolit123** [comment]
@errordeveloper 
> So what you want to add to the docs is that

can you elaborate what docs are you referring to?

kubeadm docs or k-a docs?

the k-a GCE docs already have a weave step:
https://github.com/kubernetes/kubernetes-anywhere/blob/ec6155b8a83aac93b6a3e82c830847ef64bc2ce9/phase1/gce/DEVEL.md#deploying-a-kubernetes-development-cluster

```
kubectl apply -f https://git.io/weave-kube
```
this gets 2.4.0.

**Comment by neolit123** [comment]
@fabriziopandini found a failure with the new manifest:
```
+ kubectl apply -f clusters/kubernetes/.tmp
error: error validating "clusters/kubernetes/.tmp/weave-net-list.json": error validating data: [field spec.template.spec.containers[0].securityContext.procMount for v1.SecurityContext is required, field spec.template.spec.containers[1].securityContext.procMount for v1.SecurityContext is required]; if you choose to ignore these errors, turn validation off with --validate=false
Makefile:71: recipe for target 'addons' failed
make: *** [addons] Error 1
```
investigating.

**Comment by neolit123** [comment]
now using the suggested line by @errordeveloper 
```
curl -s "https://cloud.weave.works/k8s/net.json?k8s-version=v1.10.0" | 
jq . > phase3/weave-net/weave-net-list.json
```

@fabriziopandini 
verified that this manifest works, looks like the json conversation that i did might have messed something up. this one got the manifest as `json` directly.

**Comment by neolit123** [comment]
updated this too:
> Also bump the kubectl version to 1.12.0-beta.1 as 1.6.x is ancient.

**Comment by roberthbailey** [comment]
Ping me if you need this merged before I have a chance to fix up the repository ACLs.

**Comment by neolit123** [comment]
@errordeveloper we had a discussion about kubernetes-anywhere docs at the SIG cluster lifecycle meeting today and the consensus was that _we shouldn't_ add any docs updates for this project.

the project is maintained for e2e only.

@fabriziopandini PTAL, both you and me now have merge access, thanks to Robert, but we can wait a couple of hours more if there are further comments here.

**Pull request merged**